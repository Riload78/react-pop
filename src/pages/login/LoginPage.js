import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Button from 'react-bootstrap/Button'
import Form from 'react-bootstrap/Form'
import auth from './service.js'
import { useState } from 'react'
import { useAuth } from './context.js'

const LoginPage = () => {
  const { onLogin, isSessionSaved, changeSessionStatus } = useAuth() // Obtener funciones y estado del contexto

  const [formValues, setFormValues] = useState({
    email: '',
    password: '',
  })
  const [isSave, setIsSave] = useState(true)

  const handlerChange = event => {
    setFormValues(currentFormValues => ({
      ...currentFormValues,
      [event.target.name]: event.target.value,
    }))
  }

  const handlerSwitch = event => {
    const isSaved = event.target.checked
    setIsSave(isSaved)
    changeSessionStatus(isSaved) // Cambiar el estado de la sesiÃ³n guardada
  }

  const handleSubmit = async event => {
    event.preventDefault()
    await auth.login(formValues, isSave)
    onLogin() // No es necesario pasar isSave ya que el contexto lo gestiona
  }

  const { email, password } = formValues
  return (
    <Container className='login-wrapper min-vh-100 d-flex justify-content-center'>
      <Row className='justify-content-center align-self-center g-4'>
        <Form className='login-form ' onSubmit={handleSubmit}>
          <h1>Login</h1>
          <Form.Group className='mb-3' controlId='formBasicEmail'>
            <Form.Label>Email address</Form.Label>
            <Form.Control
              type='email'
              name='email'
              value={email}
              placeholder='Enter email'
              onChange={handlerChange}
            />
            <Form.Text className='text-muted'>
              We'll never share your email with anyone else.
            </Form.Text>
          </Form.Group>

          <Form.Group className='mb-3' controlId='formBasicPassword'>
            <Form.Label>Password</Form.Label>
            <Form.Control
              type='password'
              name='password'
              value={password}
              placeholder='Password'
              onChange={handlerChange}
            />
          </Form.Group>
          <Button variant='primary' size='sm' type='submit'>
            Submit
          </Button>
        </Form>
        <Form className='login-form'>
          <Form.Check
            type='switch'
            id='custom-switch'
            label={!isSave ? "I wan't save the session" : 'Save the session'}
            checked={isSave}
            onChange={handlerSwitch}
          />
        </Form>
      </Row>
    </Container>
  )
}

export default LoginPage
