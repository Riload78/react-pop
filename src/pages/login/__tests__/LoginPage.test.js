import { render, screen } from '@testing-library/react'
import LoginPage from '../LoginPage'
import { Provider } from 'react-redux'
describe('LoginPage', () => {
    const state = {
        auth: {
            isSessionSaved: true
        },
        ui:{
            loading: false,
            notification: {
                type: null,
                message: null
            }
        }
    }

    const store = {
      dispatch: () => {},
      getState: () => state,
      subscribe: () => {},
      replaceReducer: () => {},
    }
    const renderComponent = () => render(
        <Provider store={store}>
            <LoginPage />
        </Provider>
    )

    test('snapshot', () => {
        const { container } = renderComponent()
        expect(container).toMatchSnapshot()
    })

    test('should dispach login action', () => {
        renderComponent()
        const email = screen.getByLabelText(/email/)
        const password = screen.getByLabelText('password')
        const remember = screen.getByLabelText('remember')
        const submit = screen.getByLabelText('submit')
    })
})