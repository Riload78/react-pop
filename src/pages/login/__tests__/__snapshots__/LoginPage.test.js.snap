// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `
<div>
  <div
    class="login-wrapper d-flex justify-content-center container"
  >
    <div
      class="justify-content-center align-self-center g-4 row"
    >
      <form
        class="login-form "
      >
        <h1>
          Login
        </h1>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
          >
            Email address
          </label>
          <input
            class="form-control"
            name="email"
            type="email"
            value=""
          />
          <small
            class="text-muted form-text"
          >
            We'll never share your email with anyone else.
          </small>
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
          >
            Password
          </label>
          <input
            autocomplete="password"
            class="form-control"
            name="password"
            type="password"
            value=""
          />
        </div>
        <button
          class="btn btn-primary btn-sm"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
      <form
        class="login-form"
      >
        <div
          class="form-check form-switch"
        >
          <input
            class="form-check-input"
            id="custom-switch"
            type="checkbox"
          />
          <label
            class="form-check-label"
            for="custom-switch"
            title=""
          >
            I wan't save the session
          </label>
        </div>
      </form>
    </div>
  </div>
</div>
`;
